src/QMCWaveFunctions/TrialWaveFunction.h

```diff
@@ -124,6 +124,9 @@ class TrialWaveFunction: public MPIObjectBase
   ///differential laplacians
   ParticleSet::ParticleLaplacian_t L;
 
+  ///the size of gradient component (QMCTraits::DIM)*the number of particles
+  int TotalDim;
```

Comment:
Please move these three lines back to its original place and replace all your use of TotalDim with G.size()*DIM. TotalDim is not safe to use and to be removed.